### PR TITLE
Use recent dependency versions

### PR DESCRIPTION
**Pull request by ljacomet**
Includes one API switch to resolve a deprecation warning.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Comment by ljacomet** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_7.adoc [hunk 1]
```diff
@@ -731,6 +731,12 @@ Its functionality has been merged into the `gradlePlugin` block.
 These changes require recent versions of the Plugin Publish plugin (link:https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.1.0[1.0.+]).
 Documentation on configuring plugin publication can be found both link:https://plugins.gradle.org/docs/publish-plugin[on the Portal] and <<publishing_gradle_plugins#configure_the_plugin_publishing_plugin,in the user manual>>.
 
+==== Upgrade of the version catalog parser
```
Comment (ov7a) [inline subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_7.adoc:734]: ❌ This should land to 8.x upgrade guide, right?

**Review by ov7a — CHANGES_REQUESTED** [review]

File: subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_7.adoc (hunk 1 above)
Comment (ljacomet) [reply subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_7.adoc:734]: 🤦 Right, fixing

That's the part I forgot to move when reviving this PR

Comment (ljacomet) [reply subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_7.adoc]: Fixed

**Review by ov7a — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Review by lkasso — APPROVED** [review]

**Comment by ljacomet** [comment]
No worry, will work on integrating this for 8.6 RC1 once branching is done. There is no urgency to have this merged.

**Comment by ljacomet** [comment]
@bot-gradle merge

**Comment by ljacomet** [comment]
@bot-gradle cancel

**Comment by ljacomet** [comment]
One of the updated dependency impacts performance a lot. Might need to break this in smaller PRs.

**Review by lkasso — APPROVED** [review]
Docs LGTM

**Pull request closed without merging**